### DISCUSSION TITLE
Cannot open an adios2 file transferred across machines

### DISCUSSION THREAD

**Discussion by wtaitano** [ornladios/ADIOS2#4354] [Q&A]
Dear Adios Community,

I have an adios2 file that was created from a simulation on one HPC machine. When I copy that file to another HPC machine (slightly different architecture, and compilers used) and restart the simulation, I get the following error:

`[ADIOS2 ERROR] <Helper> <adiosSystem> <ExceptionToError> : adios2_open_new_comm: [Mon Sep 16 17:43:33 2024] [ADIOS2 EXCEPTION] <Engine> <BP4Reader> <MetadataExpectedMinFileSize> : ADIOS Index file record-2.9.2.bp is assumed to always contain n*64 byte-length records. The file size now is 24838 bytes.`

Would anyone know what might be the issue? Note, I am able to open the adios file and use it to restart my simulation if I use the same HPC machine.

Please let me know if any additional info is needed.

Thank you,

-Will

**Comment by pnorbert** [comment]
You probably have `record-2.9.2.bp/mmd.0` file, which indicates that it is a BP5 file. The BP4 engine cannot process this dataset. Please make sure you have BP5 built with your adios installation on this other machine.

**Reply by wtaitano** [reply]
Thanks a lot. I just confirmed that there is an mmd.0 file, so will rebuild with BP5 and check.

**Reply by wtaitano** [reply]
I just checked the adios output files on both machines and both have mmd.0 So I presume BP5 is built with our adios on both machines. Perhaps the machine that can't open the file is trying to open as BP4? Is there a quick way to check what engine version is being used?

**Reply by pnorbert** [reply]
That you copied the mmd.0 to the second machine is no proof that you have BP5 support on it. With adios 2.10, `bpls -Vv` would tell you the list of engines that are built into that installation but I am afraid that was not available in 2.9.

How did you install adios on the second machine?

**Reply by wtaitano** [reply]
Thanks for the quick response again. I was not clear; What I meant was, I ran a separate simulation on the second machine and checked files it created and it contained mmd.0. 

I'm checking with the dev that built adios on the second machine what the build options he used were. Will get back to you hopefully tomorrow.  Thanks for the help so far.

**Comment by pnorbert** [comment]
If there is no such file, and indeed this is BP4 dataset, please check that you transferred the files properly and the size of each file in that .bp directory matches the original. md.0 is a table of 64 byte records, your file has 6 extra bytes somehow.

**Reply by wtaitano** [reply]
Thanks for the quick responses! Will check and get back, soon.

**Comment by eisenhauer** [comment]
Closing this as ancient.
